src/KOKKOS/fix_efield_kokkos.cpp

```diff
@@ -43,13 +44,17 @@ FixEfieldKokkos<DeviceType>::FixEfieldKokkos(LAMMPS *lmp, int narg, char **arg)
 {
   kokkosable = 1;
   atomKK = (AtomKokkos *) atom;
+  domainKK = (DomainKokkos *) domain;
   execution_space = ExecutionSpaceFromDevice<DeviceType>::space;
-  datamask_read = EMPTY_MASK;
```

Comment:
If there is only one parallel region then it is fine to set `datamask_read` to a global value, but if there are multiple parallel regions and each need different data, then it is better to set `datamask_read = EMPTY_MASK` and then sync/modify the correct data in each function instead. This avoids transferring extra data that is not needed. So this change needs to be reverted.

Reply:
see if you like this commit https://github.com/lammps/lammps/pull/4190/commits/88a32e691638c93bfec22fa7d59f5d75224484bb 

By refactoring Domain and DomainKokkos with some additional utility methods and using polymorphism, fix_deform_kokkos.cpp is down to 9 lines of code and resilient to future changes in base class.

The reason for this bug was base class filtered by groupbit and kokkos class did not for x2lamda() and lamda2x() in end_of_step().